src/software/ai/hl/stp/stp.cpp

```diff
@@ -137,17 +145,58 @@ std::vector<std::shared_ptr<Tactic>> STP::assignRobotsToTactics(
     // https://github.com/saebyn/munkres-cpp is the implementation of the Hungarian
     // algorithm that we use here
 
-    if (world.friendlyTeam().numRobots() < tactics.size())
+    auto friendly_team         = world.friendlyTeam();
+    auto& friendly_team_robots = friendly_team.getAllRobots();
+
+    // Special handling for the Goalie, since there should only ever be one and only
+    // one robot per team is ever permitted to be one
+    const std::optional<Robot> goalie = friendly_team.goalie();
+    // Assign the goalie to the first goalie tactic
+    auto it = tactics.begin();
+    if (goalie)
+    {
+        while (it != tactics.end())
+        {
+            if ((*it)->isGoalieTactic())
+            {
+                (*it)->updateRobot(*goalie);
+                break;
+            }
+            it++;
+        }
+    }
+
+    // Discard the goalie tactic we assigned and any other goalie tactics, since
+    // there is only one goalie
+    while (it != tactics.end())
```

Comment:
consider using `remove_if` https://en.cppreference.com/w/cpp/algorithm/remove

Reply:
Change this and removed most manual iterator usage usage above as well! Can't believe I forgot about this................